Automattic/jetpack
# Update/statuses to make more sense with ownership

## Proposed changes:

* Update statuses to avoid showing the "Learn More/Inactive" state when a product is "owned"
* If product is owned, allow activation if there is a free offering that does not require a plan
* If product is owned but requires a plan (even a free one) show a "Needs plan" status with a "Get plan" CTA
* If product needs first site connection, but product is owned, show a "Needs Connection" status with the CTA of "Connect" instead of "Learn More/Inactive"

### Other information:


## Jetpack product discussion

P2: pbNhbs-b6Y-p2

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

1. Create a new JN site with Jetpack Beta installed (no Jetpack plugin)
2. Activate Jetpack Protect with this branch
3. Deactivate the Akismet plugin
4. Go to My Jetpack and confirm you see Protect with the status of "Needs Connection" and a CTA of "Connect" with Akismet being "Inactive" but with the option to "Activate"
![image](https://github.com/user-attachments/assets/dc45dd12-9602-4378-8511-37635e2f3287)
5. Activate the base Jetpack plugin with this branch and install the Jetpack VaultPress Backup standalone plugin. Make sure you see the new "Needs Plan" status with the CTA of "Get plan"
![image](https://github.com/user-attachments/assets/c7c60b28-9c1a-42f3-9926-87aadfeb9207)
6. Install the Boost plugin and make sure you see the option to Activate Boost in My Jetpack
![image](https://github.com/user-attachments/assets/755241f2-2e77-4aa0-b865-9ffcc143cecb)
7. Connect your site and make sure the statuses look correct
![image](https://github.com/user-attachments/assets/00979636-e77c-414a-9c80-21c648f63282)
8. Now connect your user and make sure they still look correct
![image](https://github.com/user-attachments/assets/29f20ca3-460a-4e9a-a90a-88fbdf375714)
9. Uninstall Jetpack Protect and add a paid plan for it from SA and make sure you see the "Needs plugin" status
![image](https://github.com/user-attachments/assets/dbcefcab-0b73-4019-a3c5-06e5a5ef93df)
10. Try activating some of the plugins (such as Boost) and make sure they remain active
![image](https://github.com/user-attachments/assets/674c972b-9532-46b4-9cfe-c7f33917e38a)
11. Try disconnecting your user account and activating a plugin that requires a user connection and make sure the card updates to needing a connection
![image](https://github.com/user-attachments/assets/a02066e8-0d39-41b6-9674-99d978f23008)


Statuses have a lot of different scenarios, so I can't account for all of them here so feel free to play around and try out anything you can think of
